tensorflow/compiler/xla/service/llvm_ir/llvm_target_helper.h

```diff
@@ -0,0 +1,50 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_COMPILER_XLA_SERVICE_LLVM_IR_LLVM_TARGET_HELPER_H_
+#define TENSORFLOW_COMPILER_XLA_SERVICE_LLVM_IR_LLVM_TARGET_HELPER_H_
+
+#include <string>
+#include "llvm/ADT/Triple.h"
+#include "llvm/IR/Intrinsics.h"
+#include "llvm/IR/Module.h"
+
+namespace xla {
+namespace llvm_ir {
+// Enmerations to get target specific intrinsics or function calls
+enum TargetIntrinsicID {
+  kShfl_down_f32,
+  kShfl_down_i32,
+  kThread_id_x,
+  kThread_id_y,
+  kThread_id_z,
+  kBlock_id_x,
+  kBlock_id_y,
+  kBlock_id_z,
+  kBarrier_id,
+  kLast_id
+};
+
+struct TargetIntrinsics {
+  llvm::Intrinsic::ID nvptx_intrinsic;
+  llvm::Intrinsic::ID amdgpu_intrinsic;
+};
+
+typedef struct TargetIntrinsics GPUIntrinsics; 
```

Comment:
...I think I see why you're doing this, but using a typedef doesn't make a ton of sense to me.  Should this file just be in xla::gpu instead of xla::llvm_ir?

Reply:
Do you have a suggestion where it should reside ? I do anticipate there may be more of such constants that need to be maintained  such as for accessing calling conventions, target specific address spaces etc.

Reply:
A new file under xla/gpu?  Maybe `target_utils.h`?  Kind of generic.  `target_abstractions.h`? 
 (And since EmitCallToTargetIntrinsic is really about GPU code, it should go in the same file.)